java/lib/src/main/java/com/svix/Endpoint.java

```diff
@@ -13,97 +13,121 @@
-		}
-	}
-
-	public void rotateSecret(final String appId, final String endpointId, final EndpointSecretRotateIn endpointSecretRotateIn) throws ApiException {
-		try {
-			api.rotateEndpointSecretApiV1AppAppIdEndpointEndpointIdSecretRotatePost(endpointId, appId, endpointSecretRotateIn, null);
-		} catch (com.svix.internal.ApiException e) {
-			throw Utils.wrapInternalApiException(e);
-		}
-	}
-
-	public void recover(final String appId, final String endpointId, final RecoverIn recoverIn) throws ApiException {
-		try {
-			api.recoverFailedWebhooksApiV1AppAppIdEndpointEndpointIdRecoverPost(appId, endpointId, recoverIn, null);
-		} catch (com.svix.internal.ApiException e) {
-			throw Utils.wrapInternalApiException(e);
-		}
-	}
-
-	public EndpointHeadersOut getHeaders(final String appId, final String endpointId) throws ApiException {
-		try {
-			return api.getEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersGet(endpointId, appId, null);
-		} catch (com.svix.internal.ApiException e) {
-			throw Utils.wrapInternalApiException(e);
-		}
-	}
-
-	public void updateHeaders(final String appId, final String endpointId, final EndpointHeadersIn endpointHeadersIn) throws ApiException {
-		try {
-			api.updateEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPut(appId, endpointId, endpointHeadersIn, null);
-		} catch (com.svix.internal.ApiException e) {
-			throw Utils.wrapInternalApiException(e);
-		}
-	}
-
-	public void patchHeaders(final String appId, final String endpointId, final EndpointHeadersIn endpointHeadersIn) throws ApiException {
-		try {
-			api.patchEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPatch(appId, endpointId, endpointHeadersIn, null);
-		} catch (com.svix.internal.ApiException e) {
-			throw Utils.wrapInternalApiException(e);
-		}
-	}
+    private final EndpointApi api;
+
+    public Endpoint() {
+        api = new EndpointApi();
+    }
+
+    public ListResponseEndpointOut list(final String appId, final EndpointListOptions options) throws ApiException {
+        try {
+            return api.listEndpointsApiV1AppAppIdEndpointGet(appId, options.getIterator(), options.getLimit(), null);
+        } catch (com.svix.internal.ApiException e) {
+            throw Utils.wrapInternalApiException(e);
+        }
+    }
+
+    public EndpointOut create(final String appId, final EndpointIn endpointIn) throws ApiException {
+        try {
+            return api.createEndpointApiV1AppAppIdEndpointPost(appId, endpointIn, null);
+        } catch (com.svix.internal.ApiException e) {
+            throw Utils.wrapInternalApiException(e);
+        }
+    }
+
+    public EndpointOut create(final String appId, final EndpointIn endpointIn, final PostOptions options) throws ApiException {
+        try {
+            return api.createEndpointApiV1AppAppIdEndpointPost(appId, endpointIn, options.getIdempotencyKey());
+        } catch (com.svix.internal.ApiException e) {
+            throw Utils.wrapInternalApiException(e);
+        }
+    }
```

Comment:
Rather than duplicating all the logic in these functions it would be much cleaner to make your

`create(final String appId, final EndpointIn endpointIn)` function call your `create(final String appId, final EndpointIn endpointIn, final PostOptions options)` function (just without an idempotency key.

you should do this for all of the overloaded functions